sklearn/_build_utils/deprecated_modules.py

```diff
@@ -270,9 +270,22 @@
 
 _FILE_CONTENT_TEMPLATE = """
 # THIS FILE WAS AUTOMATICALLY GENERATED BY deprecated_modules.py
+from . import {new_module_name}
+
+if {new_module_name}.__file__.endswith(".py"):
+    if not hasattr({new_module_name}, '__all__'):
+        {new_module_name}.__all__ = []
```

Comment:
maybe avoid the condition with something like `new_module.__all__ = getattr(new_module, '__all__', [])`?